packages/vite/src/node/optimizer/optimizer.ts

```diff
@@ -614,13 +682,38 @@ async function createDepsOptimizer(
       return
     }
 
+    if (isBuild) {
+      currentlyProcessing = false
+      const crawlDeps = Object.keys(metadata.discovered)
+      if (crawlDeps.length === 0) {
+        debug?.(
+          colors.green(
+            `✨ no dependencies found while processing user modules`,
+          ),
+        )
+        firstRunCalled = true
+      } else {
+        runOptimizer()
+      }
+      return
+    }
```

Comment:
Moved the build time logic here to simplify the dev logic. The comment could also be improved and during build we don't need the debounce. We want to remove this anyways, but the change helped with this PR.